hazelcast/src/main/java/com/hazelcast/query/impl/predicates/LikePredicate.java

```diff
@@ -43,6 +47,38 @@ public LikePredicate(String attributeName, String expression) {
         this.expression = expression;
     }
 
+    @Override
+    public Set<QueryableEntry> filter(QueryContext queryContext) {
+        Index index = queryContext.getIndex(attributeName);
+        String indexPrefix = expression.substring(0, expression.length() - 1);
+        return index.getRecords(indexPrefix, true, indexPrefix + "\uFFFF", false);
+    }
+
+    @Override
+    public boolean isIndexed(QueryContext queryContext) {
+        Index index = queryContext.getIndex(attributeName);
```

Comment:
It's better to use `queryContext .matchIndex(attributeName, PREFER_ORDERED)` here to make sure we are preferring ordered indexes over the unordered ones (there might be multiple indexes for the same attribute).

Reply:
addressed in https://github.com/hazelcast/hazelcast/pull/18289/commits/3ff9fab612cdc376e0ab4300bce4650637188820